src/store.rs

```
@@ -158,13 +159,13 @@ impl Timelog {
     }
 
     pub fn save(&self) -> Result<(), io::Error> {
-        assert!(self.filename.is_some());
-        let filename = self.filename.as_ref().unwrap();
-        if let Some(parent) = filename.parent() {
-            fs::create_dir_all(parent)?;
+        if let Some(filename) = &self.filename {
```

Please revert this. If save() ever ends up being called without a file name, this is an internal programming error. Like this, save() would silently succeed without writing a file. If you need this for some purposes, please split out into a separate commit with a rationale, and return an io:Error if self.filename is not set.

---

Was not sure about that tbh. On one side i didn't wanted to be destructive but on the other side a panic can destroy data too. Reverted